plugins/out_s3/s3.c

```diff
@@ -1071,6 +1248,9 @@ static int upload_data(struct flb_s3 *ctx, struct s3_file *chunk,
     if (ctx->compression == FLB_AWS_COMPRESS_GZIP) {
         flb_free(payload_buf);
     }
+    else if (ctx->compression == FLB_AWS_COMPRESS_PARQUET) {
+        flb_sds_destroy(payload_buf);
```

Comment:
Why are we using `flb_sds_destroy` vs `flb_free`?

Reply:
This is because we create flb_sds_t here for parquet case:
https://github.com/fluent/fluent-bit/pull/8837/files#diff-dac37a34397e1fe40f475c0eb941c76063fbdee5134632dcb47d5daed77ecb4dR477